kolibri/core/device/middleware.py

```diff
@@ -33,3 +40,48 @@ def process_view(self, request, view_func, view_args, view_kwargs):
         if request.META.get("HTTP_USER_AGENT", None) == "Kolibri session":
             return HttpResponse('')
         return None
+
+
+ALLOWED_PATH_LIST = [
+    "kolibri:deviceprovision",
+    "kolibri:setupwizardplugin:setupwizard",
+    "kolibri:set_language",
+    "kolibri:session-list",
+]
```

Comment:
Instead of having this middleware with its whitelist, couldn't we just add a cached property somewhere and add the check together with the login check? Our login check happens in decorators for views.

Assumption being, that all pages that require user login also require device is provisioning.

Reply:
Not quite, there are anonymous pages that require provisioning. May be good to just have a "provisioning exempt decorator. Can revisit.

Reply:
An unprovisioned device will usually see its users display a select few pages. I know that adding middleware can give some guarantee protection (although not for management scripts), but what I thought was convenient was to avoid this whitelist and middleware simply by checking for provision in the most obvious places.

Maybe login-requiring pages + the login page itself aren't enough?

Reply:
I think no API endpoints will give horrendous responses here, so it may be easier to just blend this into the user redirection view instead (and then we can drop the middleware).

Reply:
Yes, removing the middleware to guarantee that it doesn't cloud out API endpoints is a good point.

In KA Lite, we had plenty of issues because of this type of middleware+whitelist... well, it wasn't even a list :smile: https://github.com/learningequality/ka-lite/blob/master/kalite/distributed/middleware.py#L43